teps/0154-concise-remote-resolver-syntax.md

```diff
@@ -0,0 +1,376 @@
+
+```yaml
+taskRef:
+  resolver: <resolver-name>
+  params:
+    - name: ...
+      value: ...
+    ...
+``` 
+
+For fetching the resource already applied to the cluster, we rely on the name field:
+```yaml
+taskRef:
+  name: name-of-the-task
+```
+
+Currently, users can either use the `name` field or the combination of `resolver` and `params`. The two fields (`name` and `resolver`) cannot be used together as the admission controller will not allow it.
+We propose changing this behavior such that they are both allowed. We propose updating the `name` field whose value can also be the [resolver-reference](resolver-reference). For example: 
+
+```yaml
+taskRef:
+  name: <resolver-reference>
+  resolver: <resolver-name>
+```
+
+In turn, the Tekton controller will pass the `name` to the `Remote Resolver` (identified by the `resolver` field ). The remote resolver will parse the `name` and extract the necessary information it needs to perform resolution. In [stage 2](#stage-2), Tekton will auto-detect which resolver to forward the request to so that users dont need to explicitly declare the name of the resolver on the cluster.
+ 
+#### Resolver Reference
+
+[Package-URL](https://github.com/package-url/purl-spec) is a format that we have begun to use in Tekton Artifacts. To enable a consistent concise format for across resolvers, we propose taking purl as an inspiration to define our format. The reason for not using purl directly is that it is too rigid for the different types of resolvers.
+
+To provide a consistent format across resolvers, we suggest the following `remote-resolver` template: `<scheme>://<location>[@<version>][#<selector>][?<params>]`. This template is a "best practice" so that all resolvers follow a similar template. It will not be enforced by the Tekton Pipeline/Task controllers. In the template, `scheme` is required while `location`, `version`, `selector` and `params` are all optional and should be parsed by the `Remote Resolvers` according to their needs. Additionally, the value of the `name` must be `url encoded`.
+In [stage 2](#stage-2), remote resolvers will subscribe to `scheme`s that they support and Tekton will auto-forward the resolution request to the subscribed resovers. 
+
+The `Remote Resolver` must define the format using the above `remote-resolver` template. The `Remote Resolver` is be responsible for parsing the `name` and extracting the necessary information. The Tekton controller will only validate that the `name` starts with `<scheme>://`. If a `scheme` is found, it will match against the schemes that the resolvers accept. If provided `scheme` is not supported by any of the resolvers, then it will throw an error. If the `name` does not start with `<scheme>://`, the Tekton controller with then treat it as a local reference and not forward the request to the remote resolvers. Validation of the other optional fields must be conducted by the resolvers themselves. 
```

Comment:
```suggestion
The `Remote Resolver` must define the format using the above `remote-resolver` template. The `Remote Resolver` is responsible for parsing the `name` and extracting the necessary information. The Tekton controller will only validate that the `name` starts with `<scheme>://`. If a `scheme` is found, it will match against the schemes that the resolvers accept. If provided `scheme` is not supported by any of the resolvers, then it will throw an error. If the `name` does not start with `<scheme>://`, the Tekton controller will then treat it as a local reference and not forward the request to the remote resolvers. Validation of the other optional fields must be conducted by the resolvers themselves. 
```

Reply:
Fixed